obf-generation/contours/V4/make-contour-tile-mt.sh

```diff
@@ -138,7 +138,162 @@ process_tiff ()
 		size_str=$(gdalinfo $filepath | grep "Size is" | sed 's/Size is //g')
 		width=$(echo $size_str | sed 's/,.*//')
 		height=$(echo $size_str | sed 's/.*,//')
+  
+        if [[ $filename =~ ^([NS])([0-9]+)([EW])([0-9]+)$ ]]; then
+            lat_prefix="${BASH_REMATCH[1]}"
+            lon_prefix="${BASH_REMATCH[3]}"
+            lat="${BASH_REMATCH[2]}"
+            lon="${BASH_REMATCH[4]}"
+            lat_digits="${#lat}"
+            lon_digits="${#lon}"
+        else
+            echo "Error: Filename must be in the format [NS]XX[EW]YYY.tif"
+            exit 1
+        fi
+               
+        left_exists=0
+        right_exists=0
+        bottom_exists=0
+        top_exists=0
+  
+        neighbors=()
+        for dlat in -1 0 1; do
+            for dlon in -1 0 1; do
+                new_lat=$(echo "$lat + $dlat" | bc)
+                new_lon=$(echo "$lon + $dlon" | bc)
+
+                if (( $(echo "$new_lat >= 0" | bc -l) )); then
+                    if [[ "$lat_prefix" == "N" ]]; then
+                        lat_prefix="N"
+                    else
+                        lat_prefix="S"
+                    fi
+                else
+                    if [[ "$lat_prefix" == "N" ]]; then
+                        lat_prefix="S"
+                    else
+                        lat_prefix="N"
+                    fi
+                    new_lat="${new_lat#-}"  # Remove the negative sign
+                fi
+                
+                if (( $(echo "$new_lon >= 0" | bc -l) )); then
+                    if [[ "$lon_prefix" == "E" ]]; then
+                        lon_prefix="E"
+                    else
+                        lon_prefix="W"
+                    fi
+                else
+                    if [[ "$lon_prefix" == "E" ]]; then
+                        lon_prefix="W"
+                    else
+                        lon_prefix="E"
+                    fi
+                    new_lon="${new_lon#-}"  # Remove the negative sign
+                fi
+        
+                formatted_lat=$(printf "%0${lat_digits}d" "$new_lat")
+                formatted_lon=$(printf "%0${lon_digits}d" "$new_lon")
+
+                neighbor_filename="${indir}${lat_prefix}${formatted_lat}${lon_prefix}${formatted_lon}.tif"
```

Comment:
Slash (**/**) should be added after ${indir} because earlier it is removed
`indir=${1%/*}`

Reply:
Fixed